### PR TITLE
Add Android bundle to GitHub Action build artifacts

### PR DESCRIPTION
**Pull request by buberdds**
Release task has only AAB to avoid confusion what goes to Google app store. Build workflow has both AAB and APK (for testing). Sample fork output  https://github.com/buberdds/oasis-wallet-web/actions/runs/9283327194

Alternative builds samples for future reference:
1. using `android-actions/setup-android@v3` action. Has hardcoded command line tools versions (may be outdated). 
2. manual instalation of Android command line tools. more ugly wokrflow, with hardoded url, but we control everything.

```
      - name: Download Command Line Tools
        # Expected location for sdkmanager is: <sdk>/cmdline-tools/latest/
        # CLI tools package https://developer.android.com/studio#command-line-tools-only
        run: |
          curl -o commandlinetools.zip https://dl.google.com/android/repository/commandlinetools-linux-11076708_latest.zip
          mkdir -p $HOME/android-sdk/cmdline-tools/latest
          unzip commandlinetools.zip -d $HOME/android-sdk/cmdline-tools
          mv $HOME/android-sdk/cmdline-tools/cmdline-tools/* $HOME/android-sdk/cmdline-tools/latest/
          rm -rf $HOME/android-sdk/cmdline-tools/cmdline-tools
      - name: Set up environment variables
        run: |
          echo "ANDROID_HOME=$HOME/android-sdk" >> $GITHUB_ENV
          echo "ANDROID_SDK_ROOT=$HOME/android-sdk" >> $GITHUB_ENV
          echo "PATH=$HOME/android-sdk/cmdline-tools/latest/bin:$PATH" >> $GITHUB_ENV
      - name: Accept SDK licenses
        run: yes | sdkmanager --licenses

      - name: Install SDK components
        run: |
          sdkmanager "platform-tools" "platforms;android-33" "build-tools;33.0.0"
      - name: Build Android ROSE Wallet
        run: ./gradlew bundleRelease
        working-directory: android
```

APK artifact alternative:
 - download bundletool jar end extract APK from AAB

TODO:
 - [x] build AAB file rather thank APK using `gradlew bundleRelease` instead of `assembleRelease` (actually we want both to speed up testing)
 - [x] confirm Java version used by workflow https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions. Should we use `temurin` distribution? 
 - [x] confirm Android platform and build tools versions https://github.com/oasisprotocol/oasis-wallet-web/blob/master/android/app/src/main/AndroidManifest.xml#L40, so it's 31 ? It's 34 now after switching to Capacitor v6
 - [x] choose bundle build flow
 - [x]  jarsigner
 - [x] confirm appId `com.oasisprotocol.wallet` - switch to `org.`
 - [x] update release workflow (fork sample: https://github.com/buberdds/oasis-wallet-web/releases/tag/v1.11.0)
 - [x] add GitHub secrets to repo
 - [x] add apk for testing 
 
 ### GH secrets
- generate keystore password with a password manager (e.g. `pass generate rose-wallet/android-keystore-password`)
- generate keystore file:
 ```
keytool -genkeypair -v -keystore rose-wallet-android.jks -keyalg RSA -keysize 4096 -validity 100 -alias rose-wallet-android -storepass $(pass rose-wallet/android-keystore-password)
```
when asked, enter something like below:
```
Enter the distinguished name. Provide a single dot (.) to leave a sub-component empty or press ENTER to use the default value in braces.
What is your first and last name?
  [Unknown]:  Oasis
What is the name of your organizational unit?
  [Unknown]:  Engineering
What is the name of your organization?
  [Unknown]:  Oasis Protocol Foundation
What is the name of your City or Locality?
  [Unknown]:  
What is the name of your State or Province?
  [Unknown]:  
What is the two-letter country code for this unit?
  [Unknown]:  
Is CN=Oasis, OU=Engineering, O=Oasis Protocol Foundation, L=Unknown, ST=Unknown, C=Unknown correct?
  [no]:  yes
```

- encode keystore file in base64 format so it can be added as GitHub secret
```
base64 rose-wallet-android.jks > rose-wallet-android.jks.base64
```
- provide all secrets in repo settings https://github.com/oasisprotocol/oasis-wallet-web/settings/secrets/actions -> New repository secret: `KEYSTORE_FILE` , `KEYSTORE_PASSWORD` and `KEY_ALIAS`.

### Dev after package name update (Android Studio)
- File -> Sync Project with Gradle Files
- Build -> Clean Project
- Build -> Rebuild Project.

### CODE REVIEW

**Comment by lukaw3d** [comment]
(here's what I did to try the first version https://github.com/lukaw3d/try-android-ci/blob/master/.github/workflows/ci-build.yml)

**Comment by buberdds** [comment]
thanks, will switch to AAB and use jarsigner with gh secrets.

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -42,6 +47,44 @@ jobs:
         run: yarn build
       - name: Build extension ROSE Wallet
         run: yarn build:ext
+      - name: Sync Capacitor for Android
+        run: yarn cap sync android
+      - name: Accept SDK licenses
+        run: yes | $ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager --licenses
+      - name: Install SDK components
+        run: |
+          $ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager "platform-tools" "platforms;android-31" "build-tools;31.0.0"
```
Comment (lubej) [inline .github/workflows/ci-build.yml:56]: Can the app be build with older SDK version(Since we are targeting 34)?

Comment (buberdds) [reply .github/workflows/ci-build.yml:56]: ah thanks need to sync after https://github.com/oasisprotocol/oasis-wallet-web/pull/1957

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -24,6 +24,11 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+      - name: Set up OpenJDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'adopt'
```
Comment (lubej) [inline .github/workflows/release.yml:30]: Any reason we are using different distribution under [ci-build.yml](https://github.com/oasisprotocol/oasis-wallet-web/pull/1944/files#diff-68ba9d12f6f2601fe3c9f7f9d0d02aaea52384559d6d08fae7afc941663fdfcd)

Comment (buberdds) [reply .github/workflows/release.yml]: forgot to update

**Review by tjanez — APPROVED** [review]

**Pull request merged**